src/servers/grpc_server.cc

```diff
@@ -4491,6 +4493,12 @@ GRPCServer::Start()
   model_infer_cq_ = grpc_builder_.AddCompletionQueue();
   model_stream_infer_cq_ = grpc_builder_.AddCompletionQueue();
   grpc_server_ = grpc_builder_.BuildAndStart();
+  // Check if binding port failed
+  if (grpc_server_ == nullptr) {
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_UNAVAILABLE,
+        (std::string("Port '") + server_addr_ + "' already in use ").c_str());
+  }
```

Comment:
I'm not sure this is the correct spot to check. I believe `grpc_server_ == nullptr` indicates a failure to start the server, but it didn't indicate the reason for failure.

I believe `AddListeningPort()` will specifically tell you whether we failed due to trying to bind a port that's already in use: https://github.com/triton-inference-server/server/blob/afeb2a630b83ca204f8962b832db22d06e316204/src/servers/grpc_server.cc#L4452

Reference: https://github.com/grpc/grpc/blob/1dff2dc6604f98bc09a8a3677646c196025cd475/include/grpcpp/impl/codegen/server_interface.h#L136-L148

I think that may be more correct to verify a port issue specifically.

Reply:
The GitHub documentation can be a bit misleading. The error is only returned [after BuildAndStart is called](https://grpc.github.io/grpc/cpp/classgrpc_1_1_server_builder.html#ad8c7eff4f5747333b7ffbb09cef71b5e). So it would need to be checked here.

When I moved it to this part, I dropped the bound port to avoid an unnecessary variable. You raise a great point about specificity of the error, so I pushed a change to check the whether the port was bound specifically for this error. Thanks for finding this and providing documentation!